src/main/java/org/json/JSONTokener.java

```diff
@@ -397,51 +408,95 @@ public String nextTo(String delimiters) throws JSONException {
 
 
     /**
-     * Get the next value. The value can be a Boolean, Double, Integer,
-     * JSONArray, JSONObject, Long, or String, or the JSONObject.NULL object.
-     * @throws JSONException If syntax error.
+     * Get the next value. The value can be a Boolean, Double, Integer, JSONArray, JSONObject, Long, or String, or the
+     * JSONObject.NULL object.
      *
      * @return An object.
+     * @throws JSONException If syntax error.
      */
     public Object nextValue() throws JSONException {
+        return nextValue(new JSONParserConfiguration());
+    }
+
+    /**
+     * Get the next value. The value can be a Boolean, Double, Integer, JSONArray, JSONObject, Long, or String, or the
+     * JSONObject.NULL object. The strictMode parameter controls the behavior of the method when parsing the value.
+     *
+     * @param jsonParserConfiguration which carries options such as strictMode and allowSingleQuotes, these methods will
+     *                                strictly adhere to the JSON syntax, throwing a JSONException for any deviations.
+     * @return An object.
+     * @throws JSONException If syntax error.
+     */
+    public Object nextValue(JSONParserConfiguration jsonParserConfiguration) throws JSONException {
         char c = this.nextClean();
         switch (c) {
-        case '{':
-            this.back();
-            try {
-                return new JSONObject(this);
-            } catch (StackOverflowError e) {
-                throw new JSONException("JSON Array or Object depth too large to process.", e);
-            }
-        case '[':
-            this.back();
-            try {
-                return new JSONArray(this);
-            } catch (StackOverflowError e) {
-                throw new JSONException("JSON Array or Object depth too large to process.", e);
-            }
+            case '{':
+                this.back();
+                return getJsonObject(jsonParserConfiguration);
```

Comment:
Please revert code in this method that is not needed for this feature. I.E. lines 435-440, except for passing the JsonParserConfiguration object to the JSONObject and JSONArray ctors.

Reply:
Done.